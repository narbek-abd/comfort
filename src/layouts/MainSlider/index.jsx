import React from "react";
import { Swiper, SwiperSlide } from "swiper/react";
import { Pagination } from "swiper";

import "swiper/css";
import "swiper/css/pagination";
import {
	SlideLink,
	SMainSlider,
	SMiniTitle,
	STitle,
	SSubTitle,
} from "./style";

import mainImg from "../../assets/img/main.jpg";
import { SContainer, SButton } from "../../globalStyle";

const MainSlider = () => {
	return (
		<SMainSlider>
			<Swiper
				modules={[Pagination]}
				spaceBetween={50}
				slidesPerView={1}
				pagination={{ clickable: true }}
			>
				<SwiperSlide>
					<SlideLink bg={mainImg} href="#">
						<SContainer>
							<SMiniTitle>
								Best Furniture For Your Castle....
							</SMiniTitle>
							<STitle>
								New Furniture Collection Trends in 2020
							</STitle>
							<SSubTitle>
								Lorem ipsum dolor sit amet, consectetur
								adipiscing elit. Magna in est adipiscing in
								phasellus non in justo.
							</SSubTitle>
						</SContainer>
					</SlideLink>
				</SwiperSlide>

				<SwiperSlide>
					<SlideLink bg={mainImg} href="#">
						<SContainer>
							<SMiniTitle>
								Best Furniture For Your Castle....
							</SMiniTitle>
							<STitle>
								New Furniture Collection Trends in 2020
							</STitle>
							<SSubTitle>
								Lorem ipsum dolor sit amet, consectetur
								adipiscing elit. Magna in est adipiscing in
								phasellus non in justo.
							</SSubTitle>
						</SContainer>
					</SlideLink>
				</SwiperSlide>

				<SwiperSlide>
					<SlideLink bg={mainImg} href="#">
						<SContainer>
							<SMiniTitle>
								Best Furniture For Your Castle....
							</SMiniTitle>
							<STitle>
								New Furniture Collection Trends in 2020
							</STitle>
							<SSubTitle>
								Lorem ipsum dolor sit amet, consectetur
								adipiscing elit. Magna in est adipiscing in
								phasellus non in justo.
							</SSubTitle>
						</SContainer>
					</SlideLink>
				</SwiperSlide>

				<SwiperSlide>
					<SlideLink bg={mainImg} href="#">
						<SContainer>
							<SMiniTitle>
								Best Furniture For Your Castle....
							</SMiniTitle>
							<STitle>
								New Furniture Collection Trends in 2020
							</STitle>
							<SSubTitle>
								Lorem ipsum dolor sit amet, consectetur
								adipiscing elit. Magna in est adipiscing in
								phasellus non in justo.
							</SSubTitle>
						</SContainer>
					</SlideLink>
				</SwiperSlide>

				<SwiperSlide>
					<SlideLink bg={mainImg} href="#">
						<SContainer>
							<SMiniTitle>
								Best Furniture For Your Castle....
							</SMiniTitle>
							<STitle>
								New Furniture Collection Trends in 2020
							</STitle>
							<SSubTitle>
								Lorem ipsum dolor sit amet, consectetur
								adipiscing elit. Magna in est adipiscing in
								phasellus non in justo.
							</SSubTitle>
						</SContainer>
					</SlideLink>
				</SwiperSlide>
			</Swiper>
		</SMainSlider>
	);
};

export default MainSlider;
