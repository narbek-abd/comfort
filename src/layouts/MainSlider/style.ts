import styled from "styled-components";

export const MainSlider = styled.section`
margin-top: 20px;

.swiper-horizontal > .swiper-pagination-bullets .swiper-pagination-bullet
{
	width: 10px;
height: 9.89px;
background: var(--color-pink);
transform: rotate(-45deg);
border-radius: 0;
margin: 5.5px 5.5px;
}
`; 

export const SlideLink = styled.a<{bg: string}>`
display: block;
	background: url(${props => props.bg}) no-repeat center center;
	background-size: cover;
	height: 764px;
`; 



export const MiniTitle = styled.div`
padding-top: 203px;
font-family: Lato;
font-weight: bold;
font-size: 16px;
line-height: 28px;
color: var(--color-pink);

@media only screen and (max-width:768px) {
padding-top: 50px;
}
`; 
export const Title = styled.div`
font-family: Josefin Sans;
font-weight: bold;
font-size: 53px;
line-height: 70px;
letter-spacing: 0.015em;
color: #000000;
max-width: 654px;
margin-top: 12px;
@media only screen and (max-width:576px) {
font-size: 43px;
line-height: 50px;
}
`; 
export const SubTitle = styled.div`
font-family: Lato;
font-style: normal;
font-weight: bold;
font-size: 16px;
line-height: 28px;
color: #8A8FB9;
max-width: 560px;
margin-top: 31px;
`; 

