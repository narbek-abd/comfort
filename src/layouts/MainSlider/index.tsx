import React from "react";
import { Swiper, SwiperSlide } from "swiper/react";
import { Pagination } from "swiper";

import "swiper/css";
import "swiper/css/pagination";
import * as S from "./style";

import mainImg from "../../assets/img/main.jpg";
import * as G from "../../globalStyle";

const MainSlider: React.FC = () => {
	return (
		<S.MainSlider>
			<Swiper
				modules={[Pagination]}
				spaceBetween={50}
				slidesPerView={1}
				pagination={{ clickable: true }}
			>
				<SwiperSlide>
					<S.SlideLink bg={mainImg} href="#">
						<G.Container>
							<S.MiniTitle>
								Best Furniture For Your Castle....
							</S.MiniTitle>
							<S.Title>
								New Furniture Collection Trends in 2020
							</S.Title>
							<S.SubTitle>
								Lorem ipsum dolor sit amet, consectetur
								adipiscing elit. Magna in est adipiscing in
								phasellus non in justo.
							</S.SubTitle>
						</G.Container>
					</S.SlideLink>
				</SwiperSlide>

				<SwiperSlide>
					<S.SlideLink bg={mainImg} href="#">
						<G.Container>
							<S.MiniTitle>
								Best Furniture For Your Castle....
							</S.MiniTitle>
							<S.Title>
								New Furniture Collection Trends in 2020
							</S.Title>
							<S.SubTitle>
								Lorem ipsum dolor sit amet, consectetur
								adipiscing elit. Magna in est adipiscing in
								phasellus non in justo.
							</S.SubTitle>
						</G.Container>
					</S.SlideLink>
				</SwiperSlide>

				<SwiperSlide>
					<S.SlideLink bg={mainImg} href="#">
						<G.Container>
							<S.MiniTitle>
								Best Furniture For Your Castle....
							</S.MiniTitle>
							<S.Title>
								New Furniture Collection Trends in 2020
							</S.Title>
							<S.SubTitle>
								Lorem ipsum dolor sit amet, consectetur
								adipiscing elit. Magna in est adipiscing in
								phasellus non in justo.
							</S.SubTitle>
						</G.Container>
					</S.SlideLink>
				</SwiperSlide>

				<SwiperSlide>
					<S.SlideLink bg={mainImg} href="#">
						<G.Container>
							<S.MiniTitle>
								Best Furniture For Your Castle....
							</S.MiniTitle>
							<S.Title>
								New Furniture Collection Trends in 2020
							</S.Title>
							<S.SubTitle>
								Lorem ipsum dolor sit amet, consectetur
								adipiscing elit. Magna in est adipiscing in
								phasellus non in justo.
							</S.SubTitle>
						</G.Container>
					</S.SlideLink>
				</SwiperSlide>

				<SwiperSlide>
					<S.SlideLink bg={mainImg} href="#">
						<G.Container>
							<S.MiniTitle>
								Best Furniture For Your Castle....
							</S.MiniTitle>
							<S.Title>
								New Furniture Collection Trends in 2020
							</S.Title>
							<S.SubTitle>
								Lorem ipsum dolor sit amet, consectetur
								adipiscing elit. Magna in est adipiscing in
								phasellus non in justo.
							</S.SubTitle>
						</G.Container>
					</S.SlideLink>
				</SwiperSlide>
			</Swiper>
		</S.MainSlider>
	);
};

export default MainSlider;
