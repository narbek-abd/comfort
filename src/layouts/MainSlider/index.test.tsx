import { screen, waitFor, render } from "@testing-library/react";
import MainSlider from "./index";

describe("MainSlider component", () => {

	test("should display component", async () => {
		const {container} = render(<MainSlider />);

		expect(container).toMatchSnapshot(); 
	});
});
