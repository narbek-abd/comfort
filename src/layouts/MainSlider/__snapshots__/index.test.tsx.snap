// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainSlider component should display component 1`] = `
<div>
  <section
    class="sc-bdvvtL fxAsie"
  >
    <div
      class="swiper swiper-initialized swiper-horizontal swiper-pointer-events"
    >
      <div
        class="swiper-pagination swiper-pagination-clickable swiper-pagination-bullets swiper-pagination-horizontal"
      />
      <div
        class="swiper-wrapper"
        style="transition-duration: 0ms; transform: translate3d(NaNpx, 0px, 0px);"
      >
        <div
          class="swiper-slide"
        >
          <a
            class="sc-gsDKAQ cFWpXN"
            href="#"
          >
            <div
              class="sc-jRQBWg iwMRJr"
            >
              <div
                class="sc-dkPtRN laWALu"
              >
                Best Furniture For Your Castle....
              </div>
              <div
                class="sc-hKwDye eVwihf"
              >
                New Furniture Collection Trends in 2020
              </div>
              <div
                class="sc-eCImPb fRjHc"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Magna in est adipiscing in phasellus non in justo.
              </div>
            </div>
          </a>
        </div>
        <div
          class="swiper-slide"
        >
          <a
            class="sc-gsDKAQ cFWpXN"
            href="#"
          >
            <div
              class="sc-jRQBWg iwMRJr"
            >
              <div
                class="sc-dkPtRN laWALu"
              >
                Best Furniture For Your Castle....
              </div>
              <div
                class="sc-hKwDye eVwihf"
              >
                New Furniture Collection Trends in 2020
              </div>
              <div
                class="sc-eCImPb fRjHc"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Magna in est adipiscing in phasellus non in justo.
              </div>
            </div>
          </a>
        </div>
        <div
          class="swiper-slide"
        >
          <a
            class="sc-gsDKAQ cFWpXN"
            href="#"
          >
            <div
              class="sc-jRQBWg iwMRJr"
            >
              <div
                class="sc-dkPtRN laWALu"
              >
                Best Furniture For Your Castle....
              </div>
              <div
                class="sc-hKwDye eVwihf"
              >
                New Furniture Collection Trends in 2020
              </div>
              <div
                class="sc-eCImPb fRjHc"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Magna in est adipiscing in phasellus non in justo.
              </div>
            </div>
          </a>
        </div>
        <div
          class="swiper-slide"
        >
          <a
            class="sc-gsDKAQ cFWpXN"
            href="#"
          >
            <div
              class="sc-jRQBWg iwMRJr"
            >
              <div
                class="sc-dkPtRN laWALu"
              >
                Best Furniture For Your Castle....
              </div>
              <div
                class="sc-hKwDye eVwihf"
              >
                New Furniture Collection Trends in 2020
              </div>
              <div
                class="sc-eCImPb fRjHc"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Magna in est adipiscing in phasellus non in justo.
              </div>
            </div>
          </a>
        </div>
        <div
          class="swiper-slide"
        >
          <a
            class="sc-gsDKAQ cFWpXN"
            href="#"
          >
            <div
              class="sc-jRQBWg iwMRJr"
            >
              <div
                class="sc-dkPtRN laWALu"
              >
                Best Furniture For Your Castle....
              </div>
              <div
                class="sc-hKwDye eVwihf"
              >
                New Furniture Collection Trends in 2020
              </div>
              <div
                class="sc-eCImPb fRjHc"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Magna in est adipiscing in phasellus non in justo.
              </div>
            </div>
          </a>
        </div>
      </div>
    </div>
  </section>
</div>
`;
